lib/authentication/screens/signIn/services/signInController.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import 'package:home_trainer/authentication/screens/signIn/utilities/signInInteractiveTexts.dart';
+import 'package:home_trainer/authentication/services/authenticationController.dart';
+import 'package:home_trainer/authentication/screens/signIn/utilities/signInButtons.dart';
+import 'package:home_trainer/authentication/screens/signIn/utilities/signInForm.dart';
+
+class SignInController extends StatelessWidget {
+  final SignInTexts _signInTexts = new SignInTexts();
+
+  final SignInForm _emailForm = new SignInForm(
+    labelText: 'email',
+    inputType: TextInputType.emailAddress,
+    obscureText: false,
+    validator: (value) {
+      if (value.isEmpty) {
+        return 'An email is required';
+      }
+      if (!RegExp(r"^[a-zA-Z0-9.a-zA-Z0-9.!#$%&'*+" +
```

Comment:
this validation should be extracted in a different file

Reply:
Should I put every validation in separate files or make a single file with all the validations?